### ISSUE TITLE
BS breaks arbitrary strings containing hostname

### ISSUE THREAD

**Issue by OnkelTem** [BrowserSync/browser-sync#540]
This is follow-up of #528 (as it is closed now).

It seems that BS substitutes hostname not only in hostname part or URLs but also in the paths. 

For example, this was in my html:
```jade
link(href="/sites/default/themes/mclinic/css/styles.css", media="screen", rel="stylesheet")
```
and this is how it looks after BS processing:
```html
<link href="/sites/default/themes/localhost:3000/css/styles.css" media="screen" rel="stylesheet">
```
Virtual hostname was 'mclinic', Drupal theme name also 'mclinic' and thus repeated in the path. As you see BS has replaced irrelevant directory name.

Yesterday I've discovered another unexpected effect of the wrong substitution. Here is a screenshot from Drupal admin page, note the labels in the second column — "localhost:3000e_2", "localhost:3000_2v2":

![](https://sc-cdn.scaleengine.net/i/e34d16caf122d4bcfbfcdf17d1eab55a.png)

they was actually just "mclinice_2" and "mclinic_2v2" which was also replaced. Compare with the same page when no BS proxying:

![](https://sc-cdn.scaleengine.net/i/49a87d59aada8e188c08baef3b593e0c.png)

It looks like BS is replacing hostnames throughout the stream. Now wonder it leads to unpredictable results. For example, my Drupal admin section is functioning only on half :)

**Comment by shakyShane** [comment]
Bug in the proxy lib 
https://github.com/shakyShane/foxy/blob/master/lib/utils.js#L61-L120

**Comment by shakyShane** [comment]
Fixed in 2.7.9 :)

**Comment by OnkelTem** [comment]
Nice!
